lib/context.js

```diff
@@ -318,14 +325,13 @@ api.createTermDefinition = (activeCtx, localCtx, term, defined) => {
           {code: 'invalid IRI mapping', context: localCtx});
       }
       mapping['@id'] = id;
+      // indicate if this term may be used as a compact IRI prefix
+      mapping._prefix = (!mapping._termHasColon &&
+        id.match(/[:\/\?#\[\]@]$/) &&
```

Comment:
If this is in a hot path then maybe the regex should be compiled elsewhere.  Or maybe js runtimes are smart enough to do that themselves.